File: sd/cliente.py
import requests

URL = 'http://localhost:8080/funcionarios'

def listar_funcionarios():
    response = requests.get(URL)
    if response.ok:
        funcionarios = response.json()
        if not funcionarios:
            print("\nNenhum funcionário cadastrado.\n")
        else:
            print("\nLista de funcionários:")
            for f in funcionarios:
                print(f"ID: {f.get('id', '?')}, Nome: {f.get('nome')}, Cargo: {f.get('tipo')}")
            print()
    else:
        print("Erro ao listar:", response.status_code)

def cadastrar_funcionario():
    nome = input("Digite o nome do funcionário: ").strip()
    cargo = input("Digite o cargo do funcionário: ").strip()
 
    funcionario = {
        "nome": nome,
        "id": 1,
        "tipo": cargo
    }

    response = requests.post(URL, json=funcionario)
    if response.ok:
        print("\nFuncionário cadastrado com sucesso!\n")
    else:
        print("\nErro ao cadastrar:", response.status_code)

def menu():
    while True:
        print("=== Menu ===")
        print("1. Listar funcionários")
        print("2. Cadastrar novo funcionário")
        print("3. Sair")

        escolha = input("Escolha uma opção: ").strip()

        if escolha == "1":
            listar_funcionarios()
        elif escolha == "2":
            cadastrar_funcionario()
        elif escolha == "3":
            print("Saindo...")
            break
        else:
            print("Opção inválida. Tente novamente.\n")

if __name__ == "__main__":
    menu()

File: sd/cliente.js
const fetch = require('node-fetch');
const readline = require('readline');

const URL = 'http://localhost:8080/funcionarios';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function menu() {
  console.log('=== Menu ===');
  console.log('1. Listar funcionários');
  console.log('2. Cadastrar novo funcionário');
  console.log('3. Sair');
  rl.question('Escolha uma opção: ', async (opcao) => {
    switch (opcao.trim()) {
      case '1':
        await listarFuncionarios();
        menu();
        break;
      case '2':
        await cadastrarFuncionario();
        menu();
        break;
      case '3':
        console.log('Saindo...');
        rl.close();
        break;
      default:
        console.log('Opção inválida.\n');
        menu();
    }
  });
}

async function listarFuncionarios() {
  try {
    const res = await fetch(URL);
    if (res.ok) {
      const funcionarios = await res.json();
      if (!funcionarios.length) {
        console.log('\nNenhum funcionário cadastrado.\n');
      } else {
        console.log('\nLista de funcionários:');
        funcionarios.forEach(f => {
          console.log(`ID: ${f.id ?? '?'}, Nome: ${f.nome}, Cargo: ${f.tipo ?? f.cargo}`);
        });
        console.log();
      }
    } else {
      console.log('Erro ao listar:', res.status);
    }
  } catch (err) {
    console.log('Erro de conexão:', err.message);
  }
}

async function cadastrarFuncionario() {
  return new Promise((resolve) => {
    rl.question('Digite o nome do funcionário: ', (nome) => {
      rl.question('Digite o cargo do funcionário: ', async (cargo) => {
        const funcionario = {
          nome: nome.trim(),
          cargo: cargo.trim()
        };
        try {
          const res = await fetch(URL, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(funcionario)
          });
          if (res.ok) {
            console.log('\nFuncionário cadastrado com sucesso!\n');
          } else {
            console.log('\nErro ao cadastrar:', res.status);
          }
        } catch (err) {
          console.log('Erro de conexão:', err.message);
        }
        resolve();
      });
    });
  });
}

menu();
